src/umpire/resource/HipDeviceMemoryResource.cpp

```diff
@@ -24,31 +24,59 @@ HipDeviceMemoryResource::HipDeviceMemoryResource(Platform platform, const std::s
 void* HipDeviceMemoryResource::allocate(std::size_t bytes)
 {
   int old_device;
-  hipGetDevice(&old_device);
-  if (old_device != m_traits.id)
-    hipSetDevice(m_traits.id);
+  hipError_t err = hipGetDevice(&old_device);
+  if (err != hipSuccess) {
+    UMPIRE_ERROR(runtime_error,
```

Comment:
I think this message is a little misleading in that it shows an argument being passed to hipGetDevice when really the argument is written to. There's a better error message in one of the other files: `UMPIRE_ERROR(runtime_error, fmt::format("Error when trying to get HIP Device: {}", hipGetErrorString(err)));`